apps/src/localization/Localization.ts

```diff
@@ -0,0 +1,305 @@
+import {get} from 'js-cookie';
+
+import type {
+  LocalizeJS,
+  LocalizeOptions,
+} from '@cdo/apps/localization/Localize';
+import experiments from '@cdo/apps/util/experiments';
+import getScriptData from '@cdo/apps/util/getScriptData';
+import {DefaultLocale} from '@cdo/generated-scripts/sharedConstants';
+
+declare global {
+  interface Window {
+    LocalizeLoader: Promise<LocalizeJS>;
+  }
+}
+
+export type TranslatableHash = {[key: string]: string};
+
+/**
+ * The possible input types for translation. We can translate just raw strings, a list
+ * of such strings, a piece of the document as a DOM element, or a key-value store.
+ */
+export type Translatable = string[] | string | HTMLElement | TranslatableHash;
+
+export type TranslationCallbackData = {
+  code: string;
+  rtl: boolean;
+};
+
+export type TranslationCallback = (info: TranslationCallbackData) => void;
+
+/**
+ * Describes an available language.
+ */
+export type LanguageInfo = {
+  /**
+   * The localized name of the language. "English" or "Español", for example.
+   */
+  text: string;
+  /**
+   * The language code. "en" or "hi-IN", for example.
+   */
+  value: string;
+  /**
+   * Whether or not it is right-to-left.
+   */
+  rtl: boolean;
+};
+
+/**
+ * This class handles our dynamic localization engine.
+ */
+export class Localization {
```

Comment:
This class is highly specific to the Code.org Dashboard implementation right? The marketing sites implementation would be closer to the default scripts provided by LocalizeJS?